include/seqan3/io/sequence_file/format_fasta.hpp

```diff
@@ -329,47 +326,39 @@ class format_fasta
         if (options.fasta_blank_before_id)
             stream_it = ' ';
 
-        std::ranges::copy(id, stream_it);
+        stream_it.write_range(id);
 
         detail::write_eol(stream_it, options.add_carriage_return);
     }
 
     //!\brief Implementation of writing the sequence.
-    template <typename stream_it_t,
-              typename seq_type>
-    void write_seq(stream_it_t & stream_it,
-                   sequence_file_output_options const & options,
-                   seq_type && seq)
+    template <typename stream_it_t, typename seq_type>
+    void write_seq(stream_it_t & stream_it, sequence_file_output_options const & options, seq_type && seq)
     {
+        using distance_type = decltype(std::ranges::distance(seq));
+
+        auto v = seq | views::to_char;
+
         if (options.fasta_letters_per_line > 0)
         {
-        #if SEQAN3_WORKAROUND_VIEW_PERFORMANCE
-            size_t count = 0;
-            for (auto c : seq)
+            auto it = v.begin();
+            distance_type num_lines = std::ranges::distance(seq) / options.fasta_letters_per_line;
+
+            for (distance_type c = 0; c < num_lines; ++c)
             {
-                stream_it = to_char(c);
-                if (++count % options.fasta_letters_per_line == 0)
-                    detail::write_eol(stream_it, options.add_carriage_return);
-            }
-            if (count % options.fasta_letters_per_line != 0)
+                stream_it.write_n(it, options.fasta_letters_per_line);
                 detail::write_eol(stream_it, options.add_carriage_return);
```

Comment:
Are you not writing the first line of the sequence again and again here? Were do you increment `it`?
Is splitting the sequence and writing it line-wise better than chunk | join and writing as a whole? Could be either...

Reply:
> Are you not writing the first line of the sequence again and again here? Were do you increment it?

`write_n` increments `it`.. I already thought that this might be a design flaw. I can also return the updated iterator instead?

> Is splitting the sequence and writing it line-wise better than chunk | join and writing as a whole? Could be either...

Well, your code explicitly stated that a for loop is faster than view chunk join and changing the ostreambuf iterator won't change that? So I would suggest to only keep the most efficient code even though the range version is more pretty.
I do not have time for extensive benchmarks on this :/

Reply:
> write_n increments it.. I already thought that this might be a design flaw. I can also return the updated iterator instead?

Yes, that would be better! Iterators are usually passed by copy or move, not by reference.

> Well, your code explicitly stated that a for loop is faster than view chunk join and changing the ostreambuf iterator won't change that? So I would suggest to only keep the most efficient code even though the range version is more pretty. I do not have time for extensive benchmarks on this :/

Well, the point about my code was that you *can* easily benchmark it by turning a macro on/off :nerd_face: 

If I am to benchmark this again, it would really be a lot easier, if you only replace the lines that have `copy()` with `write_range()`...